imap/imapd.c

```diff
@@ -8056,8 +8065,18 @@ static void cmd_rename(char *tag, char *oldname, char *newname, char *location,
 
     if (r) {
         prot_printf(imapd_out, "%s NO %s\r\n", tag, error_message(r));
-        // ensure the mboxlist entry gets fixed up
-        if (rename_user) mboxlist_update_full(mbentry, /*localonly*/1, /*silent*/1);
+        // ensure the mboxlist entry gets fixed up or removed
+        if (olddestmbentry) {
+            int r2 = mboxlist_update_full(olddestmbentry, /*localonly*/1, /*silent*/1);
+            if (r2)
+                xsyslog(LOG_ERR, "IOERROR: replacing old destination tombstone after rename error",
+                        "mboxname=<%s>, error=<%s>", olddestmbentry->name, error_message(r));
```

Comment:
In all four cases here should we also be logging error_message(r2) so we know why the cleanup failed?

`"mboxname=<%s>, error=<%s>, cleanuperror=<%s>", oldestmbentry->name, error_message(r), error_message(r2));`

Reply:
I was going to say yes, but... actually.  *sigh*.  error_message isn't re-entrant, you can't call it twice in a single syslog because the old one will get whacked, so I'm tempted to just do two syslogs, one for the `r` and one for the `r2`.